openapi/paths/calendar@event@create.yaml

```diff
@@ -0,0 +1,25 @@
+post:
+  operationId: calendarEvent.postCreate
+  tags: [calendar, event]
+  summary: Create a calendar event
+  description: |
+    Creates a calendar event
+  requestBody:
+    description: |
+      The event to create
+    content:
+      application/json:
```

Comment:
From testing calendar event creation requests in https://github.com/alaress/schoolbox/pull/30072: AFAICT they don't yet accept JSON.

I _think_ it'd be easy enough to add: just use `$this->request->getParsedBody()` instead of `$this->requestPost` in the controller?

Reply:
That would be neat

Reply:
@armatronic I'm unsure what you mean when you say the calendar requests don't accept JSON.

The following two example requests work as expected.  Both examples use data in JSON format - the second example specifies user, author, type, etc. as object stubs instead of integers.
```
### Create
POST https://vagrant.dev.schoolbox.cloud/calendar/event/create
Accept: application/json
Authorization: Bearer {{bearerToken}}
Content-Type: application/json

{
   "userId": 1,
   "authorId": 1,
   "allDay": 0,
   "efficacy": "self",
   "title": "Newly created event",
   "startDate": "26/03/2024 2:44pm",
   "endDate": "26/03/2024 3:44pm",
   "repeat": "weekly",
   "repeat_until": "25/04/2024",
   "type": 2,
   "location": "Location of newly created event",
   "detail": "<p>We meet at the place, then we do the things.</p>",
   "publishDate": "25/03/2024 7:44pm",
   "level": 3,
   "roles": [1,2,3],
   "campuses": [1,2],
   "folders": [83,85],
   "attendance": 1,
   "guests": "14,27",
   "maxAttendees": 5,
   "showAttendees": 1
}

### Create with:
# * user, author, type specified as object stubs instead of ID
# * guests specified as array instead of comma-separated string AND
# * using object stubs instead of just IDs
# * with datetimes in RFC339 format.
# * with allDay specified as boolean
# * with seconds appended to datetimes (seconds should be accepted but ignored)
POST https://vagrant.dev.schoolbox.cloud/calendar/event/create
Accept: application/json
Authorization: Bearer {{bearerToken}}
Content-Type: application/json

{
    "user": {"id":1},
    "author": {"id":1},
    "allDay": true,
    "efficacy": "self",
    "title": "Newly created event",
    "startDate": "2024-03-26T14:44:00+10:00:45",
    "endDate": "2024-03-26T15:44:00+10:00:51",
    "repeat": "weekly",
    "repeat_until": "25/04/2024",
    "type": {"id":2},
    "location": "Location of newly created event",
    "detail": "<p>We meet at the place, then we do the things.</p>",
    "publishDate": "2024-03-25T19:44:00+10:00",
    "level": 3,
    "roles": [{"id":1},{"id":2},{"id":3}],
    "campuses": [{"id":1},{"id":2}],
    "folders": [{"id":83},{"id":85}],
    "attendance": 1,
    "guests": [{"id":14},{"id":27}],
    "maxAttendees": 5,
    "showAttendees": 1
}
```

@armatronic Is that all that is required, or am I missing something?

Reply:
The seconds in the 2nd example should be before the time zone offset, no? Didn't know that was allowed

Reply:
That was a mistake.  Well spotted, thanks.